Moggles.Domain/Application.cs

```diff
@@ -116,6 +116,48 @@ public void ChangeEnvironmentDefaultValue(string name, bool newDefaultValue)
             env.DefaultToggleValue = newDefaultValue;
         }
 
+        public void ChangeEnvironmentPosition(string environment, bool moveToLeft, bool moveToRight)
+        {
+            DeploymentEnvironments = DeploymentEnvironments.OrderBy(e => e.SortOrder).ToList();
+            var env = DeploymentEnvironments.FirstOrDefault(e => string.Compare(e.EnvName, environment, StringComparison.OrdinalIgnoreCase) == 0);
+
+            if (env == null)
+                throw new InvalidOperationException("Environment does not exists!");
+
+            if (moveToLeft && !moveToRight)
+            {
+                MoveLeft(env);
+            }
+
+            if(moveToRight && !moveToLeft)
+            {
+                MoveRight(env);
+            }
+        }
+        private void MoveLeft(DeployEnvironment environment)
+        {
+            var index = DeploymentEnvironments.IndexOf(environment);
+            if (index > 0)
+            {
+                var leftEnvironment = DeploymentEnvironments[index - 1];
+                var sortOrder = environment.SortOrder;
+                environment.SortOrder = leftEnvironment.SortOrder;
+                leftEnvironment.SortOrder = sortOrder;
+            }
+        }
+
+        private void MoveRight(DeployEnvironment environment)
+        {
+            var index = DeploymentEnvironments.IndexOf(environment);
+            if (index < DeploymentEnvironments.Count - 1) 
```

Comment:
What does this check for? That the environment is not already on the last position?
The check can be extracted in a Local Function

Reply:
You can also add a test for this